conn.go

```diff
@@ -202,21 +202,38 @@ func ParseConfig(connString string) (*ConnConfig, error) {
 		}
 	}
 
-	var topologyKeys []string = nil
+	var topologyKeys map[int][]string = nil
 	if s, ok := config.RuntimeParams["topology_keys"]; ok {
 		delete(config.RuntimeParams, "topology_keys")
 		if tkeys, err := validateTopologyKeys(s); err == nil {
-			topologyKeys = tkeys
-		} else {
-			return nil, fmt.Errorf("invalid topology_keys: %v", err)
+			topologyKeys = make(map[int][]string)
+			for _, tk := range tkeys {
+				zones := strings.Split(tk, ":")
+				if len(zones) == 1 {
+					topologyKeys[0] = append(topologyKeys[0], zones[0])
+				} else {
+					num, err := strconv.Atoi(zones[1])
+					if err != nil {
+						return nil, fmt.Errorf("invalid PREFERENCE_VALUE")
+					}
+					if num < 1 || num > MAX_PREFERENCE_VALUE {
+						return nil, fmt.Errorf("Invalid preference value for property ", zones[0], " : ", num)
+					}
+					topologyKeys[num-1] = append(topologyKeys[num-1], zones[0])
+				}
+			}
 		}
 	}
 
 	refreshInterval := int64(REFRESH_INTERVAL_SECONDS)
 	if s, ok := config.RuntimeParams["refresh_interval"]; ok {
 		delete(config.RuntimeParams, "refresh_interval")
 		if refresh, err := strconv.Atoi(s); err == nil {
-			refreshInterval = int64(refresh)
+			if refresh >= 0 && refresh <= MAX_INTERVAL_SECONDS {
+				refreshInterval = int64(refresh)
+			} else {
+				return nil, fmt.Errorf("refresh_interval has to be >=0 and <=600")
```

Comment:
Should we return an error or set it to the default value here?

Reply:
We set it to default value in pgjdbc driver, will replicate the same here.